### ISSUE TITLE
The task celery.backend_cleanup will delete my tasks added before if have the same crontab time

### ISSUE THREAD

**Issue by linxi-1214** [celery/django-celery-beat#322]
I set my task with crontab <0, 4, *, *, *>, then restart celery. 
I find my task was deleted, then I debug the code and I find a snippet executed in `django-celery-beat/models.py`:

```
@classmethod
    def from_schedule(cls, schedule):
        spec = {'minute': schedule._orig_minute,
                'hour': schedule._orig_hour,
                'day_of_week': schedule._orig_day_of_week,
                'day_of_month': schedule._orig_day_of_month,
                'month_of_year': schedule._orig_month_of_year,
                'timezone': schedule.tz
                }
        try:
            return cls.objects.get(**spec)
        except cls.DoesNotExist:
            return cls(**spec)
        except MultipleObjectsReturned:
            cls.objects.filter(**spec).delete()
            return cls(**spec)
```
I wonder why delete the crontab if `MultipleObjectsReturned` error occurred even if the crontab is bound with other task?

**Comment by mvikharev** [comment]
+1

**Comment by izimobil** [comment]
This is a major issue indeed, just came across it recently.
Probably for historic reasons and because there is no unique constraint, schedules tables can contain duplicates.
So, schedules should not be deleted as they can be linked to other tasks, IMO the correct code should be:

```python
@classmethod
    def from_schedule(cls, schedule):
        spec = {'minute': schedule._orig_minute,
                'hour': schedule._orig_hour,
                'day_of_week': schedule._orig_day_of_week,
                'day_of_month': schedule._orig_day_of_month,
                'month_of_year': schedule._orig_month_of_year,
                'timezone': schedule.tz
                }
        try:
            return cls.objects.get(**spec)
        except cls.DoesNotExist:
            return cls(**spec)
        except MultipleObjectsReturned:
            cls.objects.filter(**spec).first()
```
I'll send a pull request ASAP.

**Comment by auvipy** [comment]
looking forward to it

**Comment by hartwork** [comment]
Can probably be closed now that #389 is merged…

**Comment by auvipy** [comment]
> Can probably be closed now that #389 is merged…

shouldn't we track https://github.com/celery/django-celery-beat/pull/269 for future?

**Comment by hartwork** [comment]
> > Can probably be closed now that #389 is merged…
> 
> shouldn't we track #269 for future?

I don't have "need" for that myself right now but if someone wants to team up on the topic, I can think my way back into it. As you like.

**Comment by auvipy** [comment]
closing this issue but going to open your PR
